### ISSUE TITLE
save SSL keys or secrets in mitmdump

### ISSUE THREAD

**Issue by Nerudifus** [mitmproxy/mitmproxy#787]
Is there a way to save the SSL keys in a form for wireshark ("(Pre)-Master-Secret")? If not is it possible to add such functionality?
For example I get an "invalid header" or "Gateway Timeout" error for some SSL sites by using mitmdump. Without mitmdump everything works ok. Without seeing both request, the original one and the one used by mitmdump it is difficult to find the problem.

**Comment by Kriechi** [comment]
Yes, you need to set the `SSLKEYLOGFILE` environment variable in your shell from which mitmdump ist started.

```bash
$ export SSLKEYLOGFILE=ssl.log
$ mitmdump
...
```

**Comment by pmorch** [comment]
And for reference, the first mention of SSLKEYLOGFILE is in 224f737646a3f9d0d6540a295524806df7ed1943 which is included in the v0.13.1 tag. So it doesn't work in the mitmproxy installed in debian stable:

```bash
> mitmproxy --version
mitmproxy 0.10.1
```
